bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/core/QuickTextSearcher.java

```diff
@@ -132,22 +154,52 @@ protected void visit(IFile f, IProgressMonitor mon) {
 			}
 
 			currentFile = f;
-			try (LineReader lr = new LineReader(new InputStreamReader(f.getContents(true), f.getCharset()), MAX_LINE_LEN)) {
-				String line = null;
-				int lineIndex = 1;
-				while ((line = lr.readLine()) != null) {
-					int offset = lr.getLastLineOffset();
-					if (checkCanceled(mon)) {
-						return;
-					}
 
-					boolean found = query.matchItem(line);
-					if (found) {
-						LineItem lineItem = new LineItem(f, line, lineIndex, offset);
-						add(lineItem);
+			try {
+				String string = toShortString(f);
+				if (string != null) {
+					// fast path:
+					Pattern lineSeparator = lineSeparators.get();
+					String[] lines = lineSeparator.splitWithDelimiters(string, -1);
+					int lineIndex = 1;
+					int offset = 0;
+					for (String line : lines) {
+						// line or separator?
+						if (line.length() > 2 || !lineSeparator.matcher(line).matches()) {
+							if (checkCanceled(mon)) {
```

Comment:
The cancel check is misplaced here it should at best be the first check in the loop.